src/plugins/console/server/plugin.ts

```diff
@@ -72,15 +70,24 @@ export class ConsoleServerPlugin implements Plugin<ConsoleSetup> {
       router,
     });
 
-    registerSpecDefinitionsRoute({ router });
+    registerSpecDefinitionsRoute({
+      router,
+      services: { specDefinitions: this.specDefinitionsService },
+    });
 
     return {
-      addExtensionSpecFilePath,
-      addProcessorDefinition,
+      addExtensionSpecFilePath: this.specDefinitionsService.addExtensionSpecFilePath.bind(
```

Comment:
Happy to go with this suggestion too! I think having the plugin's setup function show what the API is in one place seems more explicit me (because it is the last point in the plugin before we expose the API), but I like the idea of pushing the plugin idiom to sub-components like services too.

Reply:
It was just a suggestion. I saw the core team do it

https://github.com/elastic/kibana/blob/master/src/core/public/http/http_service.ts